rerun_py/src/catalog/connection_handle.rs

```diff
@@ -179,18 +180,32 @@ impl ConnectionHandle {
                 })
                 .await
                 .map_err(to_py_err)?
-                .into_inner();
+                .into_inner()
+                .data
+                .ok_or_else(|| PyValueError::new_err("missing data from response"))?
+                .decode()
+                .map_err(to_py_err)?;
 
             response
-                .id
-                .ok_or_else(|| PyValueError::new_err("received response without task id"))
+                .column_by_name("rerun_task_id")
+                .ok_or_else(|| {
+                    PyValueError::new_err("bug: rerun_task_id column not found in response")
+                })?
+                .try_downcast_array_ref::<arrow::array::StringArray>()
+                .map_err(to_py_err)?
+                .iter()
+                .map(|v| {
+                    v.map(|id| TaskId { id: id.to_owned() })
+                        .ok_or_else(|| PyValueError::new_err("bug: found null task_id"))
+                })
+                .collect::<Result<Vec<_>, _>>()
```

Comment:
Once you have a schema extension defined, you can just check whether `schema()` [contains](https://docs.rs/arrow/latest/arrow/datatypes/struct.Schema.html#method.contains`) the returned dataframe, and omit further checks (i.e. just ok()/flatten() everything as needed).